mne_bids/write.py

```diff
@@ -999,64 +986,68 @@ def make_dataset_description(path, name, data_license=None,
         If overwrite is True, provided fields will overwrite previous data.
         If overwrite is False, no existing data will be overwritten or
         replaced.
-    verbose : bool
-        Set verbose output to True or False.
+    %(verbose)s
 
     Notes
     -----
     The required field BIDSVersion will be automatically filled by mne_bids.
 
     """
-    # Put potential string input into list of strings
-    if isinstance(authors, str):
-        authors = authors.split(', ')
-    if isinstance(funding, str):
-        funding = funding.split(', ')
-    if isinstance(references_and_links, str):
-        references_and_links = references_and_links.split(', ')
-    if dataset_type not in ['raw', 'derivative']:
-        raise ValueError('`dataset_type` must be either "raw" or '
-                         '"derivative."')
-
-    fname = op.join(path, 'dataset_description.json')
-    description = OrderedDict([('Name', name),
-                               ('BIDSVersion', BIDS_VERSION),
-                               ('DatasetType', dataset_type),
-                               ('License', data_license),
-                               ('Authors', authors),
-                               ('Acknowledgements', acknowledgements),
-                               ('HowToAcknowledge', how_to_acknowledge),
-                               ('Funding', funding),
-                               ('ReferencesAndLinks', references_and_links),
-                               ('DatasetDOI', doi)])
-    if op.isfile(fname):
-        with open(fname, 'r', encoding='utf-8-sig') as fin:
-            orig_cols = json.load(fin)
-        if 'BIDSVersion' in orig_cols and \
-                orig_cols['BIDSVersion'] != BIDS_VERSION:
-            raise ValueError('Previous BIDS version used, please redo the '
-                             'conversion to BIDS in a new directory '
-                             'after ensuring all software is updated')
-        for key, val in description.items():
-            if description[key] is None or not overwrite:
-                description[key] = orig_cols.get(key, None)
-    # default author to make dataset description BIDS compliant
-    # if the user passed an author don't overwrite,
-    # if there was an author there, only overwrite if `overwrite=True`
-    if authors is None and (description['Authors'] is None or overwrite):
-        description['Authors'] = ["[Unspecified]"]
-
-    pop_keys = [key for key, val in description.items() if val is None]
-    for key in pop_keys:
-        description.pop(key)
-    _write_json(fname, description, overwrite=True, verbose=verbose)
-
-
+    with use_log_level(verbose):  # XXX: remove 0.24 release
+        # https://github.com/mne-tools/mne-python/pull/9732
```

Comment:
To avoid unnecessary code churn/`blame` issues with these it's usually better to do something like:
```
with use_log_level(...):
     _make_dataset_description(...)


def _make_dataset_description(...):
    ...
```
@alexrockhill which test was failing due to the missing `__globals__`? I want to better understand the failure, replicate locally, see if there's a simpler solution, and also maybe test my proposed solution with https://github.com/mne-tools/mne-python/pull/9735 (which would be to vendor this in this PR).

Reply:
See any of the CIs above for `mne_bids.tests.test_write.py::test_make_dataset_description`